services/QuillLMS/app/services/demo/create_admin_report.rb

```diff
@@ -87,8 +99,8 @@ def reset
     end
 
     # delete some activity sessions to make data more varied
-    all_classrooms.sample(NUMBER_OF_CLASSROOMS_TO_DESTROY_SESSIONS_FOR).each do |classroom|
-      activity_sessions_for_classroom = ActivitySession.joins(:classroom_unit).where('classroom_units.classroom_id = ?', classroom.id)
+    all_classrooms.each do |classroom|
+      activity_sessions_for_classroom = ActivitySession.joins(:classroom_unit).where('classroom_units.classroom_id = ?', classroom.id).where.not(activity_id: Activity::PRE_TEST_DIAGNOSTIC_IDS)
       number_of_sessions_to_destroy = (RANGE_OF_NUMBER_OF_SESSIONS_TO_DESTROY).to_a.sample
```

Comment:
Since we're not really using the array for the sample, `rand(range)` would be more efficient.  You could also chain all of the operations:

```
ActivitySession
  .joins(:classroom_unit)
  .where(classroom_units: {classroom_id: classroom.id})
  .where.not(activity_id: Activity::PRE_TEST_DIAGNOSTIC_IDS)
  .sample(rand(RANGE_OF_NUMBER_OF_SESSIONS_TO_DESTROY))
  .each(&:destroy)
```